kahuna/public/js/util/crop.js

```diff
@@ -21,30 +26,69 @@ cropUtil.constant('freeform', freeform);
 cropUtil.constant('cropOptions', cropOptions);
 cropUtil.constant('defaultCrop', landscape);
 
-cropUtil.factory('cropTypeUtil', ['storage', function(storage) {
-  const isValidCropType = cropType => cropOptions.some(_ => _.key === cropType);
+cropUtil.factory('cropSettings', ['storage', function(storage) {
+  function getCropOptions() {
+    const customCrop =  storage.getJs(CUSTOM_CROP_STORAGE_KEY, true);
+    return customCrop ? cropOptions.concat(customCrop) : cropOptions;
+  }
+
+  const isValidCropType = cropType => getCropOptions().some(_ => _.key === cropType);
+
+  const isValidRatio = ratio => {
+    const splitRatio = ratio.split(',');
+    return !(splitRatio.length < 3 || isNaN(splitRatio[1]) || isNaN(splitRatio[2]));
+  };
```

Comment:
We could use [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to be explicit about what `ratio` should look like. I think using the happy path (`&&`) is clearer too as it removes the global `!`:

```suggestion
  const isValidRatio = ratio => {
    const [label, x, y, ...rest] = ratio.split(',');
    return label && isNaN(x) && isNaN(y);
  };
```